### ISSUE TITLE
Windows 11 detection

### ISSUE THREAD

**Issue by animentork** [jantari/LSUClient#48]
### Version

1.3.4

### Computer Model

20U6S3XQ00

### Problem

The comparison of the installed Windows version to a driver Windows version dependency does not work as intended. Consequently, when a driver has a dependency on Windows 11, LSUClient will still install it on a Windows 10 machine. This can lead to blue screens!

In Test-MachineSatisfiesDependencies.ps1, this part of the code does not behave as you would expect:

```
        '_OS' {
            foreach ($entry in $Dependency.OS) {
                if ("$entry" -like "${CachedHardwareTable['_OS']}*") {
                    return 0
                }
            }
```

If you change the code as follows, it works:

```
        '_OS' {
            foreach ($entry in $Dependency.OS) {
                if ("$entry" -like ("{0}*" -f $CachedHardwareTable['_OS'])) {
                    return 0
                }
            }
```

You can easily try this yourself:

```
$test1 = "Win10"
$test2 = @{"_OS" = "Win10"}

"WIN11" -like "Win10*"  # ---> returns False
"WIN11" -like "${test1}*"  # ---> returns False
"WIN11" -like "${test2["_OS"]}*"  # ---> unexpectedly returns True!
"WIN11" -like ("{0}*" -f $test2["_OS"])  # ---> returns False
```

I think the asterisk * character behaves different when placed after the hashtable lookup. So I propose to change the line as shown above ort to something similar.

Great work, btw.!

**Edit: This seems to be the problem:**

```
"${test2["_OS"]}*"  # ---> returns *
"$($test2["_OS"])*"  # ---> returns Win10*
```

So you can change the line to:

`if ("$entry" -like "$($CachedHardwareTable['_OS'])*") {`

**Comment by jantari** [comment]
I'll look into this later today, thanks a lot for the detailed issue!

**Comment by jantari** [comment]
Thanks again, this will be fixed in the next version (1.4.0) which I'm hoping to release very shortly anyway - if I don't find any more problems during testing!

**Comment by jantari** [comment]
Fixed in Version 1.4.0 🎉
